server_cmds.go

```diff
@@ -15,129 +15,156 @@ type ServerMethods struct {
-	BandwidthSentLastSecond       uint64  `ms:"connection_bandwidth_sent_last_second_total"`
-	BandwidthReceivedLastSecond   uint64  `ms:"connection_bandwidth_received_last_second_total"`
-	BandwidthSentLastMinute       uint64  `ms:"connection_bandwidth_sent_last_minute_total"`
-	BandwidthReceivedLastMinute   uint64  `ms:"connection_bandwidth_received_last_minute_total"`
-	ConnectedTime                 uint32  `ms:"connection_connected_time"`
-	PacketLossTotalAvg            float32 `ms:"connection_packetloss_total"`
-	PingTotalAvg                  float32 `ms:"connection_ping"`
-	PacketsSentSpeech             uint64  `ms:"connection_packets_sent_speech"`
-	PacketsReceivedSpeech         uint64  `ms:"connection_packets_received_speech"`
-	BytesSentSpeech               uint64  `ms:"connection_bytes_sent_speech"`
-	BytesReceivedSpeech           uint64  `ms:"connection_bytes_received_speech"`
-	PacketsSentKeepalive          uint64  `ms:"connection_packets_sent_keepalive"`
-	PacketsReceivedKeepalive      uint64  `ms:"connection_packets_received_keepalive"`
-	BytesSentKeepalive            uint64  `ms:"connection_bytes_sent_keepalive"`
-	BytesReceivedKeepalive        uint64  `ms:"connection_bytes_received_keepalive"`
-	PacketsSentControl            uint64  `ms:"connection_packets_sent_control"`
-	PacketsReceivedControl        uint64  `ms:"connection_packets_received_control"`
-	BytesSentControl              uint64  `ms:"connection_bytes_sent_control"`
-	BytesReceivedControl          uint64  `ms:"connection_bytes_received_control"`
+	ConnectedTime                  time.Duration `ms:"connection_connected_time"`
+	Ping                           float64       `ms:"connection_ping"`
+	PacketlossTotal                float64       `ms:"connection_packetloss_total"`
+	FileTransferBandwidthSent      uint64        `ms:"connection_filetransfer_bandwidth_sent"`
+	FileTransferBandwidthReceived  uint64        `ms:"connection_filetransfer_bandwidth_received"`
+	FileTransferBytesSentTotal     uint64        `ms:"connection_filetransfer_bytes_sent_total"`
+	FileTransferBytesReceivedTotal uint64        `ms:"connection_filetransfer_bytes_received_total"`
+	PacketsSentTotal               uint64        `ms:"connection_packets_sent_total"`
+	PacketsReceivedTotal           uint64        `ms:"connection_packets_received_total"`
+	BytesSentTotal                 uint64        `ms:"connection_bytes_sent_total"`
+	BytesReceivedTotal             uint64        `ms:"connection_bytes_received_total"`
+	BandwidthSentLastSecond        uint64        `ms:"connection_bandwidth_sent_last_second_total"`
+	BandwidthReceivedLastSecond    uint64        `ms:"connection_bandwidth_received_last_second_total"`
+	BandwidthSentLastMinute        uint64        `ms:"connection_bandwidth_sent_last_minute_total"`
+	BandwidthReceivedLastMinute    uint64        `ms:"connection_bandwidth_received_last_minute_total"`
+}
+
+// convert duration nanoseconds to seconds
```

Comment:
This would break if called multiple times, see the discussion on decoding below.